xarray/core/accessor_str.py

```diff
@@ -76,17 +80,46 @@ class StringAccessor:
     """
 
     __slots__ = ("_obj",)
+    _pattern_type = type(re.compile(""))
 
     def __init__(self, obj):
         self._obj = obj
 
-    def _apply(self, f, dtype=None):
+    def _apply(self, f, obj=None, dtype=None, otype=None, signature=None, **kwargs):
         # TODO handling of na values ?
+        if obj is None:
+            obj = self._obj
         if dtype is None:
-            dtype = self._obj.dtype
+            dtype = obj.dtype
+        if otype is None:
+            otype = dtype
 
-        g = np.vectorize(f, otypes=[dtype])
-        return apply_ufunc(g, self._obj, dask="parallelized", output_dtypes=[dtype])
+        g = np.vectorize(f, otypes=[otype], signature=signature)
+        return apply_ufunc(g, obj, dask="parallelized", output_dtypes=[dtype], **kwargs)
```

Comment:
```suggestion
        return apply_ufunc(g, obj, dask="parallelized", output_dtypes=[dtype], vectorize=True, **kwargs)
```

`apply_ufunc` has a `vectorize` keyword. It also automatically infers the signature (given the correct `input_`- and `output_core_dims`.

Reply:
Fixed, see below.